### DISCUSSION TITLE
`--ext` CLI arg stopped working after 9.0.0 update

### DISCUSSION THREAD

**Discussion by adamlui** [eslint/eslint#18280] [General]
### Environment

Node version: 21.6.2
npm version: 10.2.4
Local ESLint version: 9.0.0
Global ESLint version: none
Operating System: Windows 10 Home 22H2

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
    'extends': ['eslint:recommended'],
    'ignorePatterns': ['**/*.min.js', '!/.github', '**/sandbox/**/*.js'],
    'rules': {
        'indent': 'off', 'no-unexpected-multiline': 'off', // allow whitespace anywhere
        'semi': ['error', 'always'], // enforce semicolons at end of statement
        'quotes': ['error', 'single'], // enforce single quotes for string literals
        'key-spacing': ['error', { 'beforeColon': false, 'afterColon': true }], // enforce spacing in object properties
        'comma-dangle': ['error', 'never'], // enforce no trailing commas in arrays or objects
        'no-async-promise-executor': 'off', // allow promise executor functions to be async (to accomodate await lines)
        'no-constant-condition': 'off', // allow constant conditions
        'no-empty': 'off', // allow empty blocks
        'no-inner-declarations': 'off', // allow function declarations anywhere
        'no-useless-escape': 'off' // allow all escape chars cause ESLint sucks at detecting truly useless ones
    },
    'parserOptions': { 'ecmaVersion': 2022, 'sourceType': 'script' },
    'overrides': [{ 'files': ['**/*.yml'], 'extends': ['plugin:yml/standard'] }],
    'env': { 'browser': true, 'node': true, 'es6': true }
};
```
</details>

```js
name: ESLint

on: [push, pull_request]

jobs:
  eslint:
    runs-on: ubuntu-latest
    steps:

      - name: Checkout repository code
        uses: actions/checkout@v2

      - name: Set up Node.js
        uses: actions/setup-node@v2
        with:
          node-version: 18

      - name: Install dependencies
        run: npm ci # instead of install to ensure consistency and determinism in CI/CD workflow

      - name: Run ESLint
        run: npm run lint
```

### What did you expect to happen?

Before bump from 8.57.0 to 9.0.0, `eslint . --ext js,yml --cache` works to lint JavaScript + YML files

### What actually happened?

After bump from 8.57.0 to 9.0.0, `eslint . --ext js,yml --cache` results in 'Invalid option '--ext' - perhaps you meant '-c'?'

https://github.com/adamlui/js-utils/actions/runs/8581136174/job/23517906860

### Link to Minimal Reproducible Example

https://github.com/adamlui/minify.js/actions/runs/8576473243/job/23507474217

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

This first occurred on GitHub Actions (Ubuntu 22.04.4 runner) after dependabot bumped eslint to 9.0.0, then `git pull` on local machine resulted in errorless same cmd, then updating node_modules contents w/ `npm i` resulted in same new error

_No response_

**Comment by fasttime** [comment]
Hi @adamlui, to use the legacy configuration and CLI options in ESLint v9, set the environment variable `ESLINT_USE_FLAT_CONFIG` to `false` as indicated in the [migration guide](https://eslint.org/docs/latest/use/migrate-to-9.0.0#-new-default-config-format-eslintconfigjs). For example in the Ubuntu shell you could run:

```sh
ESLINT_USE_FLAT_CONFIG=false eslint . --ext js,yml --cache
```

**Reply by adamlui** [reply]
sup @fasttime i renamed the https://github.com/adamlui/js-utils/blob/main/.eslintrc.js to eslint.config.js, what do I edit to move `--ext js,yml` into it? also was there a practical reason to make it no longer a dotfile (i have dozens of GitHub Actions sync workflows that ignore dotfiles now I have to add this new config file to which is quite inconvenient, making it .eslint.config.js would've been nicer)

**Reply by adamlui** [reply]
also since the error guides users to a doc link  https://eslint.org/docs/latest/use/command-line-interface that doesn't mention any of this, it should probably be updated to note `--ext` (or all CLI options if that's what you meant) is deprecated as of v9

**Reply by fasttime** [reply]
There were multiple breaking changes in ESLint v9. If you want to upgrade your config it's best to start with https://eslint.org/docs/latest/use/configure/migration-guide. It's a long read, but it contains all the information you need, including how to replace the legacy `--ext` CLI flag.

As for the issue with the config file not being ignored: you can choose a different name like .eslint.config.js if you prefer. In this case, be sure to specify the path of the config file on the command line, e.g. `eslint -c .eslint.config.js`.

Not all CLI options are deprecated: only the ones used in the legacy config system, like `--ext`, `--env`, etc. Do you mean it would be good to note this deprecation in the documentation page about the CLI options?

**Reply by adamlui** [reply]
> As for the issue with the config file not being ignored: you can choose a different name like .eslint.config.js if you prefer. In this case, be sure to specify the path of the config file on the command line, e.g. eslint -c .eslint.config.js.

Well that makes for a messier cmd, I'm just asking why it was decided to remove the dot is all (i.e. curious what the benefit if any was considering most config files are dot for beneficial reasons like designed to be easily ignored)

> Not all CLI options are deprecated: only the ones used in the legacy config system, like --ext, --env, etc. Do you mean it would be good to note this deprecation in the documentation page about the CLI options?

Indeed the deprecated args need a note about this, because users who bumped like me see an error with a link to the CLI args page, then click link > ctrl+F for '--ext' (since error says it's not recognized) > then don't realize what happened w/o deprecation note and think it's a bug

**Reply by adamlui** [reply]
I saw that most of them (except `--rulesdir` but I opened PR #18281 to fix) have the note it is eslintrc-only but even so, users should see in error that eslintrc is not supported anymore so they understand root cause

**Comment by fasttime** [comment]
> I'm just asking why it was decided to remove the dot is all (i.e. curious what the benefit if any was considering most config files are dot for beneficial reasons like designed to be easily ignored)

It seems the idea was to align ESLint with the conventions used by other projects. I found a relevant discussion here: https://github.com/eslint/rfcs/pull/9#discussion_r249959649.

**Comment by sysmat** [comment]
- you should have a migration script
- docs are not consistent: https://eslint.org/docs/latest/use/getting-started

```
npm init @eslint/config  --> it generate eslint.config.mjs
```

After running npm init @eslint/config, you’ll have an eslint.config.js file in your directory

is this a bug?

**Reply by sysmat** [reply]
after that I still get error

```
nvalid option '--ext' - perhaps you meant '-c'?
You're using eslint.config.js, some command line flags are no longer available. Please see https://eslint.org/docs/latest/use/command-line-interface for details.
```

**Reply by sysmat** [reply]
and this  npm init @eslint/config is bugish it put eslint 8.57.0, it is unprofessional

**Reply by adamlui** [reply]
@sysmat I feel your pain, I intend to stop updating because I saw the new config file format imports stuff from plugins now (instead of super easy string like before) and I really don't intend on contacting different plugin authors to find out what their different modules are called

**Reply by Rec0iL99** [reply]
Hi @sysmat, `npm init @eslint/config` generating an `eslint.config.mjs` file is not a bug. `eslint.config.js` is the new config file format that ESLint follows by default in `v9.0.0`. If you want it to generate a config file in the old format (`.eslintrc`), then please use an older version `npm init @eslint/config@0.4.6` (`1.0.0` is the latest version of [create-config](https://github.com/eslint/create-config)). 

`--ext` works only when you use the `eslintrc` mode (deprecated config system). If you want to continue using the `eslintrc` system in `v9.0.0`, then please set the environment variable `ESLINT_USE_FLAT_CONFIG` to `false` as Francesco had mentioned before.

**Reply by Rec0iL99** [reply]
> and this npm init @eslint/config is bugish it put eslint 8.57.0, it is unprofessional

I'm not able to reproduce this, running on version `1.0.0` (latest) (`npm init @eslint/config@latest`). What version of `create-config` did you use?

**Reply by sysmat** [reply]
so **eslint.config.mjs**  is the same as  **eslint.config.js**, hmmm

**Reply by sysmat** [reply]
`npm init @eslint/config@latest` this kind of work, but it install old devDependencies, very bad
